vipgo-helper.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * Prevent the maintenance_mode plugin returning a 503 HTTP status to Nagios.
+ *
+ * Maintenance_mode sets a 503 header on page requests if maintenance_mode is enabled and this leads to  Nagios
+ * reporting lots of server errors for sites that are just in maintenance_mode. This function sets the filter
+ * response that maintenance_mode uses to determine if it shoudl set teh 503 status header or not.
+ *
+ * @return bool Should maintenance_mode set a 503 header
+ */
+function wpcom_vip_maintenance_mode_do_not_respond_503_for_nagios() {
+
+	// Check if request is coming from Nagios.
+	if ( ! empty( $_SERVER['HTTP_USER_AGENT'] ) && false === strpos( $_SERVER['HTTP_USER_AGENT'], 'check_http' ) ) {
+
+		// Not a Nagios request so allow the 503 header to be set.
+		return true;
```

Comment:
We should not assume that `true` is the value which should be returned in case the request is not a nagios request. While `true` is [the default value](https://github.com/Automattic/maintenance-mode-wp/blob/3e56f60203335a25041172760222d1ebac9f8b78/maintenance-mode.php#L77), this callback is being hooked at priority of 30, which means that any other code could had already changed to to `false` for whatever reason (that's why there is the filter anyway), and we should honour that, and only enforce our `false` for nagios requests, which we are interested in.

That said, the `wpcom_vip_maintenance_mode_do_not_respond_503_for_nagios` function should read the filter value and return that one in case the request is not a nagios one.